frontend/src/components/login.js

```diff
@@ -1,44 +1,165 @@
 import { useState } from 'react';
-import { loginUser } from '../http/user'
+import { registerUser, loginUser } from '../http/user'
 
-const Login = ({ setUserState }) => {
+const Login = ({ setUserState, setMessage, setReturnError }) => {
+    //login states
     const [userInput, setUserInput] = useState('');
     const [passwordInput, setPasswordInput] = useState('');
-    const [loginError, setLoginError] = useState('');
+    //register states
+    const [userRegInput, setUserRegInput] = useState('');
+    const [passwordRegInput, setPasswordRegInput] = useState('');
+    const [emailRegInput, setEmailRegInput] = useState('');
+    //error and toggler states
+    const [errorState, setErrorState] = useState('');
+    const [showRegister, setShowRegister] = useState(false)
 
 
+    //login function
     const handleLogin = event => {
         //trim username and set to lowercase
         const userTrim = userInput.trim();
         const passTrim = passwordInput.trim();
         event.preventDefault();
         loginUser(userTrim, passTrim).then(res => {
             if (res.name) {
+                setReturnError("");
                 setUserState(res.name);
             } else {
-                setLoginError(res);
+                setErrorState(res);
             }
         });
     }
-    return (
-        <div className="col-12 offset-md-4 col-md-4">
-            <form className="card mx-auto px-4 py-2">
-                <div className="card-body">
-                    <p className="card-title lead text-center mb-4">You must login to continue</p>
-                    <div className="input mb-3">
-                        <label className="input-text pb-2" htmlFor="username">Username</label>
-                        <input type="text" id="username" name="username" className="form-control" aria-label="username-input" aria-describedby="username-input" onChange={(e) => setUserInput(e.target.value)} />
+    //register function
+    const handleRegister = async (event) => {
+        event.preventDefault();
+        const res = await registerUser(userRegInput, emailRegInput, passwordRegInput);
+        if (!res.name) {
+            setErrorState(res.error);
+            return res.error;
```

Comment:
Why is this function returning res.error?

Reply:
should it return nothing?

Reply:
Where is the return value being used?

Reply:
errorState

Reply:
To ask again maybe in a different way.... how are you using errorState here if that's what you're returning?

Reply:
oh maybe this res.error should be passed to the flash message through the returnError

Reply:
Well you're setting error state from this so that should be enough, right? You're not using this return value at all. This is what the code is doing...

callThisFunction(); // notice how no return value exists?

So **callThisFunction()** shouldn't return anything. This is a DOM event handler, you do not return value unless you want to change if the browser default behavior should happen.

Reply:
okay ill remove the rreturn